assets/js/theme/custom/mega-menu.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Open mega menu on search button click
+ */
+export default function megaMenu() {
+    const MOBILE_WIDTH = 768
+    const megaMenuElement = document.querySelector('.js-navigation-megamenu')
+    const searchButton = document.querySelector('.js-mega-menu-search')
+    const searchInputs = document.querySelectorAll('.js-search-quick')
+    const showMore = document.querySelectorAll('.js-show-more')
+    let hasAttachedEvent = false
+
+    const openDesktopMenu = () => {
+        megaMenuElement.previousElementSibling.classList.toggle('is-open')
+        megaMenuElement.classList.toggle('is-open')
+        searchInputs?.forEach((input) => {
+            input.focus()
+        })
+    }
+
+    const updateEventListeners = () => {
+        const windowWidth = window.innerWidth
+        if (searchButton) {
+            if (windowWidth >= MOBILE_WIDTH && !hasAttachedEvent) {
+                searchButton.addEventListener('click', openDesktopMenu)
+                hasAttachedEvent = true
+            }
+        }
+    }
+
+    updateEventListeners()
+
+    window.addEventListener('resize', updateEventListeners)
+
+    showMore.forEach((button) => {
+        button.addEventListener('click', () => {
+            const buttonLabels = button.querySelectorAll('span')
```

Comment:
Please use ".js-..." classname instead of selecting all `span`s
Also, why do you need multiple labels to select? Shouldn't there be one label?

Reply:
There are two labels inside the button. We toggle both span elements with the classname "u-hidden" where one gets hidden and the other appears. I didn't want to change the value by hardcoding into HTML, so that we can use {{lang}} translated strings:

<span>Show more</span>
<span class="u-hidden">Show less</span>